### PR TITLE
Fix `docs-check` job; numpy 2.x fixes

### PR DESCRIPTION
**Pull request by sk1p**
Try to be consistent with other jobs.

## Contributor Checklist:

* [ ] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [ ] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes
* [ ] I have added/updated test cases

## Reviewer Checklist:

* [ ] `/azp run libertem.libertem-live-data` passed

### CODE REVIEW

**Comment by matbryan52** [comment]
I am genuinely perplexed about this, without installing `uv` or `tox-uv`, and forcing also `tox` env recreation the pipeline is still using `uv` to install packages, and I think this is referencing a `py38` install which we never ask for.

The numba test troubles are equally odd, I tried on `ptycho` with the same python, environment (checked via diff), branch, and there is no trouble. It looks like it only affects 16 bits per pixel tests in `merlin/.../test_encode_roundtrip...`, independently of the encode/decode size. There aren't even any warnings emitted ! All I can guess is that it it due to numba caching somewhere?

**Comment by sk1p** [comment]
> I am genuinely perplexed about this, without installing `uv` or `tox-uv`, and forcing also `tox` env recreation the pipeline is still using `uv` to install packages, 

This might be because the `venv` is not forcibly removed if it already exists, and `tox-uv` would still possibly be installed in there.

> and I think this is referencing a `py38` install which we never ask for.

Yeah, that's the really weird one. The Python 3.8 is just one of the versions that is made available to the azure agents. I'm not sure yet, and I have no idea why `uv` picks that one up! Maybe because that's the "native" version of the `ubuntu:20.04` image we use as a base? 

We might want to move to 22.04 as a base image for the azure agent, which does still [support the Python versions we need, as far as I can see](https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa).

> If there is some kind of caching on the pipelines maybe good to reset everything? I don't think I have the access to do that myself.
> The numba test troubles are equally odd, I tried on `ptycho` with the same python, environment (checked via diff), numpy 2, branch, and there is no trouble. It looks like it only affects 16 bits per pixel tests in `merlin/.../test_encode_roundtrip...`, independently of the encode/decode size. There aren't even any warnings emitted ! All I can guess is that it it due to numba caching somewhere?

I didn't look further yet, but an issue is that the azure agent doesn't completely clean up the environment after a run, so there may indeed be some files that stick around (like tox environments, the venv, and the cache of pip/uv). I just re-created the containers, and I'll re-run it again, to see if that changed anything.

**Comment by matbryan52** [comment]
> Hum... I can reproduce the failures - they only happen in the non-jit case (`NUMBA_DISABLE_JIT=1` - via `tox -e numba_coverage` for example). I had to do some changes in LiberTEM itself for the offline decoding functions - probably there's an issue with changed dtype promotion rules (numba doesn't fully implement these yet, which is why they only fail with the jit disabled)

Ah I have just been running with `pytest ... -m with_numba` as I have found `tox` difficult when I try to run it myself. So the jit'd functions pass the tests but not the python-only form 🤔

File: azure-pipelines-data.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ parameters:
   - name: recreate_tox
     displayName: Recreate tox environments?
     type: boolean
-    default: false
+    default: true
```
Comment (matbryan52) [inline azure-pipelines-data.yml:24]: I changed this during testing but I think it would need to be changed back ?

Comment (sk1p) [reply azure-pipelines-data.yml]: Yeah, switched it back now. I'm also re-trying to use `uv` now, let's see how that goes...

Comment (sk1p) [reply azure-pipelines-data.yml]: Ok, that still tries to use the wrong Python version :cry: - no `uv` for `docs-check` then...

**Comment by matbryan52** [comment]
So weird that `uv` is fixated wrong python. Otherwise we'll done, LGTM!

**Review by matbryan52 — APPROVED** [review]

**Comment by sk1p** [comment]
> So weird that `uv` is fixated wrong python. Otherwise well done, LGTM!

Yeah, extra weird that it works fine for the other jobs, like `data_tests` or `check_example_notebooks` :shrug:

**Pull request merged**